### DISCUSSION TITLE
Dev Tool - Foundry

### DISCUSSION THREAD

**Discussion by gracelessmercy** [hiero-ledger/hiero-json-rpc-relay#554] [Ideas]
I understand the next release for the JSON-RPC relay will improve support for developer tools like Hardhat, Truffle, Web3.js and Ethers. 

is Foundry on Hedera's radar as well? 

https://book.getfoundry.sh/
https://github.com/foundry-rs/book
https://www.paradigm.xyz/2021/12/introducing-the-foundry-ethereum-development-toolbox
https://uniswapv3book.com/docs/introduction/dev-environment/

**Comment by Nana-EC** [comment]
Hey @gracelessmercy 
Thanks for the question and apologies on the late turn around.

Official foundry support is a desire but we haven't yet been able to turn our attention here.
However, many developers have been able to go ahead and begin to use foundry for some scenarios.
There may be limitation as it refers to system contract support as these get at EVM differences (Foundry uses a custom EVM that have no knowledge of Hedera logic).

The team hopes to turn it's attention here in the medium term to ensure official support.
Please let us know of specific bugs you're seeing and if they're simple enough we can get good turn around as part of our sprints.


Summary: Basic non system contract functionality should work, more complex scenarios to be addressed including bugs once we get there.
Thanks

**Comment by Nana-EC** [comment]
Cleaning up here. We confirmed foundry support some time back.
Users can find an example here https://github.com/hashgraph/hedera-json-rpc-relay/tree/main/tools/foundry-example
Thanks @gracelessmercy
